packages/server/src/auth/external.ts

```diff
@@ -187,10 +187,16 @@ async function verifyExternalCode(
       body: params.toString(),
     });
 
-    const tokens = await response.json();
-    return parseJWTPayload(tokens.id_token);
+    const responseBody = await response.json();
+
+    if (!response.ok) {
```

Comment:
@codyebberson Should this check go above `await response.json()`? As it is, if the response isn't valid JSON, the new error handling would be skipped.